plaso/parsers/onedrive.py

```diff
@@ -0,0 +1,467 @@
+        value.  None if the ObfuscatedSTringMap.txt file could not be found.
+    """
+    obfuscated_string_map_entry = self._GetFileEntryInSameDirectory(
+        file_entry, self._OBFUSCATED_STRING_MAP_FILE, file_system,
+        parser_mediator)
+
+    if not obfuscated_string_map_entry:
+      return None
+
+    obfuscated_string_map_file = obfuscated_string_map_entry.GetFileObject()
+
+    obfuscated_string_map = {}
+    with text_file.TextFile(obfuscated_string_map_file,
+        encoding='utf-16-le') as obfuscated_text_file:
+      for line in obfuscated_text_file.readlines():
+        cleaned_line = line.rstrip().split('\t')
+        if len(cleaned_line) != 2:
+          continue
+        if cleaned_line[0] in obfuscated_string_map:
+          continue
+        obfuscated_string_map[cleaned_line[0]] = cleaned_line[1]
+    return obfuscated_string_map
+
+  def _ParseGeneralKeyStore(self, file_entry, file_system, parser_mediator):
+    """Parses the general.keystore file.
+
+    Args:
+      file_entry (dfvfs.FileEntry): the OneDrive Log file entry.
+      file_system (dfvfs.FileSystem): the file system containing the OneDrive
+        logs.
+      parser_mediator (ParserMediator): mediates interactions between parsers
+        and other components, such as storage and dfVFS.
+
+    Returns:
+      bytes: the decoded AES key, or None if the general.keystore file could
+        not be found or parsed.
+    """
+    general_keystore_entry = self._GetFileEntryInSameDirectory(
+        file_entry, self._GENERAL_KEYSTORE_FILE, file_system, parser_mediator)
+
+    if not general_keystore_entry:
+      return None
+
+    general_keystore_file_object = general_keystore_entry.GetFileObject()
+    general_keystore_data = general_keystore_file_object.read()
+    json_data = json.loads(general_keystore_data)
+
+    if not json_data:
+      parser_mediator.ProduceExtractionWarning(
+          'Unable to parse OneDrive keystore file {0:s}'.format(
+              general_keystore_entry.name))
+      return None
+
+    if not json_data or not isinstance(json_data, list):
+      return None
+
+    key = json_data[0].get('Key', None)
+    if not key:
+      return None
+
+    decoded_key = base64.b64decode(key)
+
+    if len(decoded_key) != 32:
+      return None
+
+    return decoded_key
+
+  def _DeobfuscateStrings(self, extracted_strings, obfuscated_string_map):
+    """Deobfuscate strings extracted from OneDrive logs.
+
+    Args:
+      extracted_strings (list[str]): strings extracted from OneDrive log files.
+      obfuscated_string_map (dict[str, str]): the parsed obfuscated string map.
+
+    Returns:
+      list[str]: the list of deobfuscated strings.
+    """
+    if not obfuscated_string_map:
+      return extracted_strings
+
+    output_strings = []
+
+    for extracted_string in extracted_strings:
+      tokens = self._DELIMITERS_REGEX.split(extracted_string)
+      output_string = []
```

Comment:
rename `output_string` to something that closer resembles its purpose e.g. `output_string_parts`

Reply:
Done